### PR TITLE
ADD support to Deployment Plan

### LINKED ISSUE #96: Image Streamer Deployment Plan Support
### Scenario/Intent
Use the Deployment Plan resource from this module

### Environment Details
-**Puppet module for HPE OneView Version:** 2.2.0
-**OneView SDK Version:** 3.1.0
-**OneView Appliance Version:** 3.0
-**OneView Client API Version:** 300
-**Puppet Version:** 4.8
-**Ruby Version:** 2.2.6
-**Platform:** Windows/Linux

### Steps to Reproduce
Try to call the resource Image_streamer_deployment_plan

### Expected Result
To be able to use the resource Deployment Plan on Image Streamer

### Actual Result
Resource not implemented / Could not find the Image_streamer_deployment_plan type

### PR DESCRIPTION
**Pull request by abiliogp**
### Description
Adds support to Image Streamer Deployment Plan

### Issues Resolved
#96 

### Check List
- [x] New functionality includes testing.
  - [x] All tests pass (`$ rake test`).
- [x] New functionality has been documented in the README if applicable.
  - [x] New functionality has been thoroughly documented in the examples (please include helpful comments).
- [x] Changes are documented in the CHANGELOG.

### CODE REVIEW

File: lib/puppet/provider/image_streamer_deployment_plan/image_streamer.rb [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+  def parse_golden_image
+    return unless @data['goldenImageURI'] && !@data['goldenImageURI'].include?('/rest/')
+    golden_image_client = OneviewSDK::ImageStreamer.resource_named('GoldenImage', @client.api_version)
+    resource = golden_image_client.find_by(@client, name: @data['goldenImageURI'])
+    raise 'Golden Image has not been found in the Appliance.' if resource.empty?
+    @data['goldenImageURI'] = resource.first['uri']
+  end
+
+  def resource_name
```
Comment (fgbulsoni) [inline lib/puppet/provider/image_streamer_deployment_plan/image_streamer.rb]: This should be getting auto-set, not need for the method

File: lib/puppet/provider/image_streamer_deployment_plan/image_streamer.rb [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    return unless @data['oeBuildPlanURI'] && !@data['oeBuildPlanURI'].include?('/rest/')
+    build_plan_client = OneviewSDK::ImageStreamer.resource_named('BuildPlan', @client.api_version)
+    resource = build_plan_client.find_by(@client, name: @data['oeBuildPlanURI'])
+    raise 'Build Plan has not been found in the Appliance.' if resource.empty?
+    @data['oeBuildPlanURI'] = resource.first['uri']
+  end
+
+  def parse_golden_image
+    return unless @data['goldenImageURI'] && !@data['goldenImageURI'].include?('/rest/')
+    golden_image_client = OneviewSDK::ImageStreamer.resource_named('GoldenImage', @client.api_version)
```
Comment (fgbulsoni) [inline lib/puppet/provider/image_streamer_deployment_plan/image_streamer.rb]: `golden_image_client`? isn't this the golden image **class**?

File: lib/puppet/provider/image_streamer_deployment_plan/image_streamer.rb [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+  def exists?
+    super
+    parse_build_plan
+    parse_golden_image
+    @resourcetype.find_by(@client, @data).any?
+  end
+
+  def parse_build_plan
+    return unless @data['oeBuildPlanURI'] && !@data['oeBuildPlanURI'].include?('/rest/')
+    build_plan_client = OneviewSDK::ImageStreamer.resource_named('BuildPlan', @client.api_version)
```
Comment (fgbulsoni) [inline lib/puppet/provider/image_streamer_deployment_plan/image_streamer.rb]: client or class? Looks like class.

File: lib/puppet/provider/image_streamer_deployment_plan/image_streamer.rb [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    resource = golden_image_client.find_by(@client, name: @data['goldenImageURI'])
+    raise 'Golden Image has not been found in the Appliance.' if resource.empty?
+    @data['goldenImageURI'] = resource.first['uri']
+  end
+
+  def resource_name
+    'DeploymentPlan'
+  end
+
+  def self.resource_name
```
Comment (fgbulsoni) [inline lib/puppet/provider/image_streamer_deployment_plan/image_streamer.rb]: Should be getting auto-set, no need for the method

**Review by fgbulsoni — CHANGES_REQUESTED** [review]
Few changes

File: lib/puppet/provider/image_streamer_deployment_plan/image_streamer.rb [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+  mk_resource_methods
+
+  def exists?
+    super
+    parse_build_plan
+    parse_golden_image
+    @resourcetype.find_by(@client, @data).any?
+  end
+
+  def parse_build_plan
```
Comment (fgbulsoni) [inline lib/puppet/provider/image_streamer_deployment_plan/image_streamer.rb:31]: both `parse_build_plan` and `parse_golden_image` should likely not be needed if making proper use of the uri_parsing methods

Comment (marikrg) [reply lib/puppet/provider/image_streamer_deployment_plan/image_streamer.rb:31]: @abiliogp has created some issues to fix it later (#122 and #121)

**Review by fgbulsoni — APPROVED** [review]

**Pull request merged**